core/thread/fetch_spec.rb

```diff
@@ -29,6 +29,34 @@
     end
   end
 
+  describe 'with a block' do
+    it 'returns the value of the fiber-local variable if value has been assigned' do
+      th = Thread.new { Thread.current[:cat] = 'meow' }
+      th.join
+      th.fetch(:cat) { true }.should == 'meow'
+    end
+
+    it "returns the block value if fiber-local variable hasn't been assigned" do
+      th = Thread.new {}
+      th.join
+      th.fetch(:cat) { true }.should == true
+    end
+
+    it "does not call the block if value has been assigned" do
+      th = Thread.new { Thread.current[:cat] = 'meow' }
+      th.join
+      th.fetch(:cat) { raise "Not reached" }.should == 'meow'
```

Comment:
I wouldn't use exceptions as a side effect - handling exception in another thread is not so straightforward - it could be swallowed.

As for using `ScratchPad` or assigning a local variable would be simpler and better.

Reply:
Changed it to a local var, that looked easier than using `ScratchPad`